frontend/src/metabase/lib/formatting/link.ts

```diff
@@ -0,0 +1,100 @@
+import { isDate } from "metabase/lib/schema_metadata";
+
+import { formatValue } from "metabase/lib/formatting";
+import { formatDateTimeForParameter } from "./date";
+
+import type { DatasetColumn } from "metabase-types/api/dataset";
+
+interface TemplateForClickFormatFunctionParamsType {
+  value: string;
+  column: DatasetColumn;
+}
+
+export interface ValueAndColumnForColumnNameDate {
+  column: DatasetColumn;
+  parameterBySlug: string;
+  parameterByName: string;
+  userAttribute: string;
+}
+
+function formatValueForLinkTemplate(value: string, column: DatasetColumn) {
+  if (isDate(column) && column.unit) {
+    return formatDateTimeForParameter(value, column.unit);
+  }
+  return value;
+}
+
+export function renderLinkTextForClick(
+  template: string,
+  data: ValueAndColumnForColumnNameDate,
+) {
+  return renderTemplateForClick(
+    template,
+    data,
+    ({ value, column }: { value: string; column: DatasetColumn }) =>
+      formatValue(value, { column }),
+  );
+}
+
+export function renderLinkURLForClick(
+  template: string,
+  data: ValueAndColumnForColumnNameDate,
+) {
+  return renderTemplateForClick(
+    template,
+    data,
+    ({ value, column }: TemplateForClickFormatFunctionParamsType) => {
+      const valueForLinkTemplate = formatValueForLinkTemplate(value, column);
+      return encodeURIComponent(valueForLinkTemplate);
+    },
+  );
+}
+
+function renderTemplateForClick(
+  template: string,
+  data: ValueAndColumnForColumnNameDate,
+  formatFunction: any = ({ value }: { value: any }) => value,
```

Comment:
Should we use the same type [as line 46](https://github.com/metabase/metabase/pull/24674/files#diff-5e4244e34e4dff568df0e35b6979799049597d33fb11d958dfdda835fb01c0d9R46) (where the callback is passed), otherwise, it could be confusing since the signatures are different.

Reply:
Absolutely, thank you.